spec/controllers/api/v1/channels_controller_spec.rb

```diff
@@ -0,0 +1,194 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::ChannelsController, type: :controller do
+  let!(:organization) { create(:organization) }
+  let!(:channel) { create(:channel, organization: organization) }
+
+  context 'when token is valid' do
+    before do
+      request.headers['Token'] = organization.token
+    end
+
+    describe 'GET #show' do
+      it 'return channel' do
+        get :show, params: { id: channel.id }
+
+        json_response = JSON.parse(response.body)
+
+        expect(response).to have_http_status(200)
+        expect(json_response['id']).to eq(channel.id)
+      end
+    end
+
+    describe 'POST #create' do
+      context 'when name field is full' do
+        it 'change Channel.count' do
+          expect do
+            post :create, params: {
+              channel: {
+                name: 'Dunder Mifflin'
+              }
+            }
+          end.to change(Channel, :count).by(1)
+        end
+
+        it 'return new channel' do
+          post :create, params: {
+            channel: {
+              name: 'Dunder Mifflin'
+            }
+          }
+
+          json_response = JSON.parse(response.body)
+
+          expect(response).to have_http_status(200)
+          expect(json_response['name']).to eq('Dunder Mifflin')
+        end
+      end
+
+      context 'when name field is empty' do
+        it 'not change Channel.count' do
+          expect do
+            post :create, params: {
+              channel: {
+                name: ''
+              }
+            }
+          end.not_to change(Channel, :count)
+        end
+
+        it 'return status code 422' do
+          post :create, params: {
+            channel: {
+              name: ''
+            }
+          }
+
+          json_response = JSON.parse(response.body)
+
+          expect(response).to have_http_status(422)
+          expect(json_response).to include('errors')
+        end
+      end
+
+      context 'when organization field is empty' do
+        it 'return new channel' do
+          post :create, params: {
+            channel: {
+              name: 'Dunder Mifflin',
+              organization:''
```

Comment:
`organization:''` is this a valid param key?

Reply:
btw, what's the trick w/ this param? as i see you ignore all params except the name in your
`params.require(:channel).permit(:name)`